tests/cephadm/test_cephadm_upgrade.py

```diff
@@ -80,6 +82,24 @@ def run(ceph_cluster, **kwargs) -> int:
         # Monitor upgrade status, till completion
         orch.monitor_upgrade_status()
 
+        if config.get("verify_cephadm_containers"):
+            for node in ceph_cluster.get_nodes(ignore="client"):
+                out, err = node.exec_command(cmd="cephadm ls", sudo=True)
+                legacy_services = [x for x in loads(out) if x["style"] == "legacy"]
+                if legacy_services:
+                    log.info(
+                        f"Legacy services present in cephadm node {node.hostname} after adopting the cluster"
+                    )
+                    log.info(
+                        "Checking cluster status to ensure that the legacy services are not being inferred"
+                    )
```

Comment:
We have the same piece of code in two places... can we implement a separate method?

Reply:
Sure will implement a method and call in both the places.

Reply:
Done.